File: tests/Generator/resource/java/complex/MapDefinitionType.java
package org.typeapi.model;

import com.fasterxml.jackson.annotation.*;

@JsonClassDescription("Represents a map which contains a dynamic set of key value entries of the same type")
public class MapDefinitionType extends CollectionDefinitionType {
    private String type = "map";

    @JsonSetter("type")
    public void setType(String type) {
        this.type = type;
    }

    @JsonGetter("type")
    public String getType() {
        return this.type;
    }
}

File: tests/Generator/resource/python/complex/map_definition_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union
from .collection_definition_type import CollectionDefinitionType


# Represents a map which contains a dynamic set of key value entries of the same type
class MapDefinitionType(CollectionDefinitionType):
    pass


